benches/divan.rs

```diff
@@ -0,0 +1,103 @@
+fn main() {
+    init_tracing();
+
+    #[cfg(feature = "tracing")]
+    let _span = tracing::span!(tracing::Level::DEBUG, "BENCH EXAMPLES").entered();
+
+    // Run registered benchmarks.
+    divan::main();
+}
+
+#[cfg(feature = "tracing")]
+fn init_tracing() -> impl Drop {
+    use tracing_flame::FlameLayer;
+    use tracing_subscriber::{fmt, prelude::*};
+    let filter_layer = tracing_subscriber::EnvFilter::from_default_env();
+    let fmt_layer = fmt::Layer::default();
+    let (flame_layer, _guard) = FlameLayer::with_file("./tracing.folded").unwrap();
+
+    tracing_subscriber::registry()
+        .with(filter_layer)
+        .with(fmt_layer)
+        .with(flame_layer)
+        .init();
+
+    _guard
+}
+
+#[cfg(not(feature = "tracing"))]
+fn init_tracing() {}
+
+mod examples {
+    #[divan::bench]
+    fn parse_postal(bencher: divan::Bencher) {
+        let input = divan::black_box(include_str!(
+            "../tests/fixtures/postal_address.terminated.input.bnf"
+        ));
+
+        bencher.bench(|| {
+            input.parse::<bnf::Grammar>().unwrap();
+        });
+    }
+
+    #[divan::bench]
+    fn generate_dna(bencher: divan::Bencher) {
+        bencher
+            .with_inputs(|| {
+                let input = "<dna> ::= <base> | <base> <dna>
+            <base> ::= 'A' | 'C' | 'G' | 'T'";
+                let grammar: bnf::Grammar = input.parse().unwrap();
+                grammar
+            })
+            .bench_refs(|grammar| {
+                grammar.generate().unwrap();
+            });
+    }
```

Comment:
Do you see a performance difference when measuring per-input grammars versus a reused `black_box(grammar)` variable hoisted above `Bencher::bench`?

Reply:
as far as I can tell, the performance is similar. this benchmark could probably use more iterations to be sure, because I see a lot of noise between runs